sensor_validate.py

```diff
@@ -1,19 +1,26 @@
+spike_threshold_limit         = {
+                                  'soc'      :  0.05 ,
+                                  'current'  :  0.1  ,
+                                }
 
-def  _give_me_a_good_name(value, nextValue, maxDelta):
+def isAbnormalSpike(value, nextValue, maxDelta):
   if nextValue - value > maxDelta:
     return False
   return True
 
-def validate_soc_reading(values):
+def hasAbnormalSpike(values,criteria):
   last_but_one_reading = len(values) - 1
   for i in range(last_but_one_reading):
-    if(not _give_me_a_good_name(values[i], values[i + 1], 0.05)):
-      return False
-  return True
+    if(not isAbnormalSpike(values[i], values[i + 1], spike_threshold_limit[criteria])):
```

Comment:
Code reads like this -
if not `isAbnormalSpike` (means it is false), then `hasAbnormalSpike` is true.
It is contradictory. Maybe `isAbnormalSpike` is actually `noAbnormalSpike`